lang/csharp/src/apache/main/Generic/GenericDatumWriter.cs

```diff
@@ -132,6 +132,8 @@ protected override bool UnionBranchMatches(Schema sc, object obj)
                 case Schema.Type.Fixed:
                     //return obj is GenericFixed && (obj as GenericFixed).Schema.Equals(s);
                     return obj is GenericFixed && (obj as GenericFixed).Schema.SchemaName.Equals((sc as FixedSchema).SchemaName);
+                case Schema.Type.Logical:
+                    return (sc as LogicalSchema).LogicalType.IsInstanceOfLogicalType(obj);
```

Comment:
Can we still write when `obj` is the base type of the logical schema? In other words, instead of giving you a `DateTime`, what if I give you an `int`? I think the writers/readers should still be able to handle that. Same question applies to the other `Matches`-esque methods that were modified in the other reader and writer classes.

Reply:
You'd have to supply conversion functions for each combination of other supported types? Wouldn't this get complicated?

Reply:
i'm with @timjroberts here